### DISCUSSION TITLE
Include and Exclude URLs syntax

### DISCUSSION THREAD

**Discussion by j4r3kb** [mbnuqw/sidebery#455] [Q&A]
Srsly add some documentation (still showing TODO).
Are there any wildcards possible here, like:

- *.example.com
- example.com/*

or what?

**Comment by pintassilgo** [comment]
> Are there any wildcards possible

No. Two types are supported, pure text and regex (regular expressions). For Sidebery to know which of the two you're using, regexp rules must start and end with `/`.

Some examples:

- `google.com`: pure text, URL will match if it contains `google.com` at **any** part, like `https://www.google.com/`, `https://mail.google.com/`, `https://www.google.com.br/`, `https://www.example.com/?url=google.com`...

- `/^https://www\.google\.com/search.*123/`: regex. URL will match if it starts with `https://www.google.com/search` and if there's `123` at any point after that, like `https://www.google.com/searchabc123def` and `https://www.google.com/search123`.

Regex is much more powerful than match patterns ("wildcards"), but also more complicated.  In the example above:

- `^` means "exactly at the beginning" (not in the middle)
- `.` means any char
- `*` means the previous char 0 or >0 times (so `.*` is similar to a wildcard)
- `\` is to treat the next char literally (so you use `\.` to refer to the simple dot, because as already said `.` is a special class that means "any char").

**Reply by j4r3kb** [reply]
One more question, can I use multiple URLs entering one per line?

**Reply by walderf** [reply]
this still isn't documented anywhere, as far as i can tell.  heh.

**Reply by dreirund** [reply]
> *No. Two types are supported, pure text and regex (regular expressions). For Sidebery to know which of the two you're using, regexp rules must start and end with `/`.*

Somehow I could not get it to work.

I have created a container with the name "Reddit".

I have enabled the "Acces website data" permission.

For the "Reddit" container, I have switched "Include URLs" to on.

In the field below, I tried the following entries, one by one (not all at the same time):

* `reddit.com`
* `/reddit/`
* `/reddit\.com/`
* `/^.*reddit\.com.*/`

I created a new tab, typed "reddit.com" in the URL bar, which navigated me to "https://reddit.com". But the tab stayed open in the temporary container it was assigned to by the "Temprary Containers" addon, it was not (re)opened in the "Reddit" tab.

For now, I use the extension "Containerise" to oben tabs in containers (it also allows to specify regexp patterns, and also wildcard patterns), but it does not allow to specify exclusions.

I also tried with the "Temprary Containers" add on disabled. This leaves "https://reddit.com" open in no container, also not opening it in the "Reddit" container.

What might I do wrong or is wrong?

**Reply by totoroot** [reply]
@dreirund For me it also no longer works. Not sure if it broke with the latest Firefox update or I misconfigured Sidebery on the new host.

**Reply by slrslr** [reply]
Can you please mention universal sub/domain regex ?

Like I have tried:
/^(https?:\/\/)?(www\d*\.|([a-zA-Z0-9-]+\.))(webshare.cz)/
(https?:\/\/)?(www\d*.|([a-zA-Z0-9-]+.))(webshare.cz)
^https?:\/\/www\d*\.webshare\.cz
/^https?:\/\/www\d*\.webshare\.cz/

but it failed to open in the container, i have been trying to match:
http://webshare.cz*
https://webshare.cz*
http://anything.webshare.cz*
https://anything.webshare.cz*

https://regex101.com/

**Reply by pintassilgo** [reply]
Example:

`/^https?:\/\/([^/]*\.)?webshare\.cz\//`

![image](https://github.com/mbnuqw/sidebery/assets/5483864/0a54f5cb-c12b-4784-8e90-2029ca1bb73a)

**Comment by pintassilgo** [comment]
Yes.

**Comment by dreirund** [comment]
> *Example:*
> 
> *`/^https?:\/\/([^/]*\.)?webshare\.cz//`*

@pintassilgo, does this work for you in sidebery?

**Reply by pintassilgo** [reply]
Yes, it does. I must say I use Sidebery 4.10.2, not 5, but unless v5 has some bug in parsing regex it should work the same.

**Reply by pintassilgo** [reply]
You should try escaping the last slash, like this:

`/^https?:\/\/([^/]*\.)?webshare\.cz\//`
